ACE/ace/SOCK_Dgram.cpp

```diff
@@ -672,7 +672,63 @@ ACE_SOCK_Dgram::make_multicast_ifaddr (ip_mreq *ret_mreq,
       // This port number is not necessary, just convenient
       ACE_INET_Addr interface_addr;
       if (interface_addr.set (mcast_addr.get_port_number (), net_if) == -1)
-        return -1;
+        {
+#ifdef ACE_WIN32
+          IP_ADAPTER_ADDRESSES tmp_addrs;
+          // Initial call to determine actual memory size needed
+          DWORD dwRetVal;
+          ULONG bufLen = 0;
+          if ((dwRetVal = ::GetAdaptersAddresses (AF_INET,
+                                                  0,
+                                                  0,
+                                                  &tmp_addrs,
+                                                  &bufLen)) != ERROR_BUFFER_OVERFLOW)
+            return -1; // With output bufferlength 0 this can't be right.
+
+          // Get required output buffer and retrieve info for real.
+          PIP_ADAPTER_ADDRESSES pAddrs;
+          char *buf;
```

Comment:
initialise with 0

Reply:
I'll just initialize during the the actual cast and remove this line.